lib/adapters/test/playwright.ts

```diff
@@ -0,0 +1,137 @@
+import type {TestCase, TestResult} from '@playwright/test/reporter';
+
+import {PlaywrightTestResultAdapter, ImageTitleEnding, type PlaywrightAttachment} from '../test-result/playwright';
+import {UNKNOWN_ATTEMPT} from '../../constants';
+
+import type {TestAdapter, CreateTestResultOpts} from './';
+import type {ReporterTestResult} from '../test-result';
+import type {AssertViewResult, ImageFile, RefImageFile} from '../../types';
+import type {ImageDiffError} from '../../errors';
+
+export type PwtRawTest = {
+    file: string;
+    browserName: string;
+    title: string;
+    titlePath: string[];
+}
+
+export class PlaywrightTestAdapter implements TestAdapter {
+    private _test: PwtRawTest;
+
+    static create<T extends PlaywrightTestAdapter>(this: new (test: PwtRawTest) => T, test: PwtRawTest): T {
+        return new this(test);
+    }
+
+    constructor(test: PwtRawTest) {
+        this._test = test;
+    }
+
+    get original(): PwtRawTest {
+        return this._test;
+    }
+
+    get id(): string {
+        return this._test.title;
+    }
+
+    get pending(): boolean {
+        return false;
+    }
+
+    get disabled(): boolean {
+        return false;
+    }
+
+    get silentlySkipped(): boolean {
+        return false;
+    }
+
+    get browserId(): string {
+        return this._test.browserName;
+    }
+
+    get fullName(): string {
+        return this._test.title;
+    }
+
+    get file(): string {
+        return this._test.file;
+    }
+
+    get titlePath(): string[] {
+        return this._test.titlePath;
+    }
+
+    createTestResult(opts: CreateTestResultOpts): ReporterTestResult {
+        const {status, attempt = UNKNOWN_ATTEMPT, assertViewResults = []} = opts;
+
+        const testCase = {
+            titlePath: () => ['', this._test.browserName, this._test.file, ...this._test.titlePath],
+            title: this._test.title,
+            annotations: [],
+            location: {
+                file: this._test.file
+            },
+            parent: {
+                project: () => ({
+                    name: this._test.browserName
+                })
+            }
+        } as unknown as TestCase;
+
+        const attachments = assertViewResults.map(assertViewResult => {
```

Comment:
Here I need to correctly handle `assertViewResults` and convert it to attachments